Source/DafnyCore/AST/DafnyAst.cs

```diff
@@ -101,7 +103,25 @@ public class Program : INode {
     }
 
     public IToken GetFirstTopLevelToken() {
-      return DefaultModuleDef.GetFirstTopLevelToken();
+      if (DefaultModule.RootToken.Next == null) {
+        return null;
+      }
+
+      var firstToken = DefaultModule.RootToken.Next;
+      var prevFile = "";
+      // We skip all included files
+      while (firstToken != null && firstToken.Next != null && firstToken.Next.Filename != DefaultModule.RootToken.Filename) {
```

Comment:
Because the first token assigned to `DefaultModule.RootToken` is curiously not necessarily the first token of the file being parsed, but tokens coming from other included files. Since we don't want to reprint the other included files, this is the pass that skips the included files.